v2/common/src/main/java/com/google/cloud/teleport/v2/transforms/FileFormatFactory.java

```diff
@@ -0,0 +1,124 @@
+
+    public abstract String outputFileFormat();
+
+    @Override
+    public POutput expand(PCollection<KV<String, String>> records) {
+        POutput output = null;
+        WriteToGCSUtility.FileFormat outputFileFormat;
+        try {
+            outputFileFormat = WriteToGCSUtility.FileFormat.valueOf(outputFileFormat());
+        } catch (Exception e) {
+            LOG.info(
+                    "Requested output file format is not a valid argument. File format needs to be either "
+                            + Arrays.asList(WriteToGCSUtility.FileFormat.values()).toString());
+            throw new IllegalArgumentException(
+                    "Requested output file format is not a valid argument. File format needs to be either "
+                            + Arrays.asList(WriteToGCSUtility.FileFormat.values()).toString());
+        }
+        /*
+         * Calls appropriate Class Builder to performs PTransform based on user provided File Format.
+         */
+        switch (outputFileFormat) {
+            case TEXT:
+                output =
+                        records.apply(
+                                "Write Text File(s)",
+                                WriteToGCSText.newBuilder()
+                                        .withOutputDirectory(options().getOutputDirectory())
+                                        .withOutputFilenamePrefix((options().getOutputFilenamePrefix()))
+                                        .withNumShards(options().getNumShards())
+                                        .withTempLocation(options().getTempLocation())
+                                        .build());
+                break;
+
+            case AVRO:
+                output =
+                        records.apply(
+                                "Write Avro File(s)",
+                                WriteToGCSAvro.newBuilder()
+                                        .withOutputDirectory(options().getOutputDirectory())
+                                        .withOutputFilenamePrefix((options().getOutputFilenamePrefix()))
+                                        .withNumShards(options().getNumShards())
+                                        .withTempLocation(options().getTempLocation())
+                                        .build());
+                break;
+
+            case PARQUET:
+                output =
+                        records.apply(
+                                "Write Parquet File(s)",
+                                WriteToGCSParquet.newBuilder()
+                                        .withOutputDirectory(options().getOutputDirectory())
+                                        .withOutputFilenamePrefix((options().getOutputFilenamePrefix()))
+                                        .withNumShards(options().getNumShards())
+                                        .build());
+                break;
+
+            default:
+                break;
```

Comment:
what happens when you get to this line?  "output" will be null which we will return.

Reply:
Yes. And didn't add any Exceptions since already we are checking the file format before entering the switch case.

Reply:
LOG.info(errorMessage);
throw new IllegalArgumentException(errorMessage);

Added these statements.

Reply:
We would still want to throw an IllegalArgumentException or UnsupportedOperationException in the default case. The corner case here is that if a new file format is added to the FileFormat enum but is not added here due to oversight.